File: algorithms/1.swift
// https://leetcode.com/problems/two-sum/

class Solution {
    func twoSum(nums: [Int], _ target: Int) -> [Int] {
        // lookup + build map along the way
        var map = [Int: Int]()
        for i1 in 0..<nums.count {
            if let i2 = map[target - nums[i1]] {
                return [i1, i2]
            } else {
                map[nums[i1]] = i1
            }
        }
        return []
    }
}

File: algorithms/1.py
# https://leetcode.com/problems/two-sum/


class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        map = {}
        for i1 in range(len(nums)):
            diff = target - nums[i1]
            if diff in map:
                return [i1, map[diff]]
            else:
                map[nums[i1]] = i1
        return []
